src/ucp/core/ucp_am.c

```diff
@@ -969,6 +923,27 @@ ucp_am_params_check_memh(const ucp_request_param_t *param, uint32_t *flags_p)
     return UCS_OK;
 }
 
+static int
+ucp_am_send_nbx_check_header_length(ucp_worker_h worker, size_t header_length)
+{
+    ucp_worker_attr_t attr;
+
+    attr.field_mask = UCP_WORKER_ATTR_FIELD_MAX_AM_HEADER;
+    if (ucp_worker_query(worker, &attr) != UCS_OK) {
+        ucs_error("failed to query maximum allowed header size");
+        return 0;
+    }
+
+    if (header_length > attr.max_am_header) {
+        ucs_error("active message header length (%zi) is greater than maximum "
+                  "allowed header size (%zi)",
+                  header_length, attr.max_am_header);
+        return 0;
```

Comment:
since this function prints an error, it should also return status code (not 1/0)
we can make it inline, and do something like:
```c
if (ENABLE_PARAMS_CHECK && (header_length > worker->max_am_header)) {
    ucs_error(...);
    return UCS_ERR_INVALID_PARAM;
}

return UCS_OK;
```

```c
status = ucm_am_check_header_length(..);
if (status != UCS_OK) {
    ret = UCS_STATUS_PTR(status);
    goto out;
}
```

Reply:
fixed